src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,39 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RuntimeException("User can not be: " + user);
+        }
+        if (user.getLogin() == null) {
+            throw new RuntimeException("User login can not be: " + user.getLogin());
+        }
+        if (user.getAge() == null) {
+            throw new RuntimeException("User age can not be: " + user.getAge());
+        }
+        if (user.getPassword() == null) {
+            throw new RuntimeException("User password can not be: " + user.getPassword());
+        }
+        if (Storage.people.contains(user.getLogin())) {
```

Comment:
you can change it to: 'if (storageDao.get(user.getLogin()) != null)'
because you have storageDao and you dont use it